src/game/phase.h

```diff
@@ -2,28 +2,40 @@
 
 #include "events.h"
 #include "construct_types.h"
+#include "context.h"
 #include <iostream>
 
 class Phase {
 public:
+    Phase(Context context) : context(context) {};
     virtual void setup() = 0;
     virtual Phase* update() = 0;
     virtual void teardown() = 0;
+    virtual void client_setup() = 0;
+    virtual void client_teardown() = 0;
+protected:
+    Context context;
 };
 
 class MenuPhase : public Phase {
 public:
+    MenuPhase(Context context) : Phase(context) {};
```

Comment:
Hmm, gives me a compilation error without - I think by default it doesn't know to call the base class's constructor.